pkg/dia/helpers/horizonhelper/stellarAssetInfo.go

```diff
@@ -0,0 +1,145 @@
+package horizonhelper
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/diadata-org/diadata/pkg/dia"
+	"github.com/sirupsen/logrus"
+	"github.com/stellar/go/clients/horizonclient"
+)
+
+const tomlReaderTimeout = 5
+
+// StellarToml represents the structure of a Stellar.toml file
+type StellarToml struct {
+	CURRENCIES []struct {
+		Code            string `toml:"code"`
+		Issuer          string `toml:"issuer"`
+		Name            string `toml:"name"`
+		DisplayDecimals int64  `toml:"display_decimals"`
+	} `toml:"CURRENCIES"`
+}
+
+func GetStellarAssetInfo(client *horizonclient.Client, assetCode, assetIssuer, blockchain string) (dia.Asset, error) {
+	assetRequest := horizonclient.AssetRequest{
+		ForAssetIssuer: assetIssuer,
+		ForAssetCode:   assetCode,
+	}
+
+	assetUrl, _ := assetRequest.BuildURL()
+	log := logrus.New().
+		WithField("context", "StellarTomlReader").
+		WithField("assetCode", assetCode).
+		WithField("assetIssuer", assetIssuer)
+
+	log.WithField("assetUrl", assetUrl).Info("input assetCode")
+
+	page, err := client.Assets(assetRequest)
+	if err != nil {
+		log.WithError(err).Error("failed to fetch asset info from horizon")
+		return dia.Asset{}, err
+	}
+
+	if len(page.Embedded.Records) == 0 {
+		err = errors.New("no toml file")
+		log.WithError(err).Error("failed to fetch asset info from horizon")
+		return dia.Asset{}, err
+	}
+
+	tomlURL := page.Embedded.Records[0].Links.Toml.Href
+	if tomlURL == "" {
+		err = errors.New("empty asset toml url from horizon")
+		log.WithError(err).Error("failed to fetch asset info with empty toml url")
+		return dia.Asset{}, err
+	}
+
+	log = log.WithField("tomlURL", tomlURL)
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
```

Comment:
It is necessary to take into account that some TOML files for Stellar assets are uploaded by their issuers onto servers without a certificate, or with an invalid certificate. Such assets will be skipped and not stored in the DIA database. 
It is possible to accept these TOML files; to do this, you need to specify a parameter `InsecureSkipVerify: true` in the TLS client. However, this parameter will not be accepted by your pull request validator **DeepSource**.